gwr/tests/test_sel_bw.py

```diff
@@ -9,6 +9,7 @@
 import pickle as pk
```

Comment:
Yes, they're only included for the fixed bandwidths. They could be included in the adaptive case. Should I add them? 

The only one that changes significantly is the BIC. As noted in the first linked private notebook which considers the objective functions for the model in this test, this is because the BIC for fixed gaussian is a global objective. If we change the upper bound from half the max to twice the max, it'll grow substantially, since it just wants to make the bandwidth hit that bound. 

I think the illustration for the objective surface in the first notebook, test `test_golden_fixed_BIC` will make this more clear?

Reply:
I guess it depends on whether or not we will use scipy as the default for both fixed and adaptive? You mentioned its faster for fixed but slower than adaptive. We could use custom golden section for adaptive and scipy for fixed. 

Got it, not that I understand the above changes to the intial min/max, I see how that change can propagate out into the final solution, typically only resulting a small change, but in a larger one for the BIC. Good to have a record of this, because it marks a shift away from the tests being validated against other software output.

Reply:
Your call on the defaults! I really don't mind if scipy is never default. Personally, I think we should *exclusively* use scipy, so that we can just remove all our code & tune to using the "right" ecosystem tool.

Reply:
that said, if I know the option is there, I'll just use that option!

Reply:
Can't argue with that logic. Do you think forcing rounding would significantly speed up scipy adaptive? In would be great if the default was the right tool and was the fastest. We've often recommended the adaptive as the default so it would be a bummer if it wasn't actually faster than other implementations. Either way, I guess it would be good to have tests for the adaptive scipy, if we are going to be exploring ways to optimize it.